src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php

```diff
@@ -322,6 +330,42 @@ protected function insert_attachment( $request ) {
 		);
 	}
 
+	/**
+	 * Determines the featured media based on a request param.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param int $featured_media Featured Media ID.
+	 * @param int $post_id        Post ID.
+	 * @return bool|WP_Error Whether the post thumbnail was successfully deleted, otherwise WP_Error.
+	 */
+	protected function handle_featured_media( $featured_media, $post_id ) {
+		$post_type         = get_post_type( $post_id );
+		$thumbnail_support = current_theme_supports( 'post-thumbnails', $post_type ) && post_type_supports( $post_type, 'thumbnail' ) || 'revision' === $post_type;
+
+		if ( ! $thumbnail_support && 'attachment' === $post_type && get_post_mime_type( $post_id ) ) {
```

Comment:
Won't the post type always be attachment?

Reply:
Hmm I suppose it is, perhaps unless someone is extending this controller for their post type or so.

Reply:
Removed now 👍